### ISSUE TITLE
[BUG] Pure Wayland fails without Xwayland

### ISSUE THREAD

**Issue by nahsheef** [fltk-rs/fltk-rs#1372]
## Describe the bug
Startup fails in pure Wayland session without Xwayland. Upstream https://github.com/fltk/fltk/commit/0f41797b7ab0ee010e31659f60cafd38affac8f5 built with `./autogen.sh --enable-wayland --disable-x11` works, probably since https://github.com/fltk/fltk/commit/a23c77c5a14395db48be9afd3142a8913d767f90 (https://github.com/fltk/fltk/pull/586).

## To Reproduce
```
[dependencies.fltk]
version = "1.3.33"
features = ["use-wayland"]
```

```
$ cargo run
Can't open display:
```

## Expected behavior
Start and show application window

## Desktop info
 - OS: Arch Linux
 - sway
 - wayland 1.21.0
 - cairo 1.17.8
 - libx11 1.8.4

## Additional info
Alternatively building with `CFLTK_WAYLAND_ONLY=1`, I get missing symbol linker errors in `Fl_X11_Screen_Driver.cxx.o`.

**Comment by MoAlyousef** [comment]
Thanks for the report. I’m getting a different build error when specifying CFLTK_WAYLAND_ONLY=1, which happens in cfltk code. 
I think I have a fix for that which I might try tomorrow. 
Can you try with `version =  "=1.3.28"`?
And can you post the output of `CFLTK_WAYLAND_ONLY=1 cargo build -vv`.

**Comment by nahsheef** [comment]
Basically multiple screens of unresolved `XFooBar` function symbols during `ld` step, such as

```
error: undefined symbol: XGrabPointer
>>> referenced by Fl_X11_Screen_Driver.cxx
>>> Fl_X11_Screen_Driver.cxx.o:(Fl_X11_Screen_Driver::grab(Fl_Window*))[...]3c.rlib
```

- XFlush
- XBell
- XParseGeometry
- XFreePixmap
- XWindowEvent
- XGetWindowAttributes
- XInternAtom
- XChangeProperty
- XQueryTree
- XFree
- ...

**Comment by nahsheef** [comment]
Same result with `version = "=1.3.28"`

**Comment by MoAlyousef** [comment]
Do you have CMake?
It seems the build is grabbing the prebuilt bîary since it can’t find cmake or git.

**Comment by nahsheef** [comment]
I'll retry 1.3.33 after installing CMake

**Comment by MoAlyousef** [comment]
1.3.33 currently doesn’t build with pure wayland due to another issue. Try 1.3.28

**Comment by nahsheef** [comment]
Will do

**Comment by nahsheef** [comment]
For a future build, can I enable `use-ninja` or would that have an adverse effect?

**Comment by nahsheef** [comment]
Would it make sense to replace `CFLTK_WAYLAND_ONLY` with a cargo feature similar to `no-images`?

**Comment by nahsheef** [comment]
Nice. After installing CMake, `CFLTK_WAYLAND_ONLY=1 cargo run` works and window is displayed. Trying without `CFLTK_WAYLAND_ONLY` now

**Comment by nahsheef** [comment]
Confirmed to work without `CFLTK_WAYLAND_ONLY=1` and also with `use-ninja` feature, all after installing cmake package first. شكرًا لك

**Comment by MoAlyousef** [comment]
العفو
The latest push should fix things with version 1.3.34. 
CFLTK_WAYLAND_ONLY could be made into a feature, I just didn't want feature proliferation since it's unlikely to build fltk for pure wayland. The direction upstream is to build hybrid FLTK apps by default. The plan here for version 2 is to follow upstream, and use something like only-x11 and only-wayland as features.

**Comment by nahsheef** [comment]
78977090a9c5905c2a55fd1dce381ab749a1da0b works

I don't need CFLTK_WAYLAND_ONLY and only tested it as a workaround, and I agree that hybrid mode is a better default since it avoids carrying two executables.
